network/ClientNetworking.cpp

```diff
@@ -127,13 +128,23 @@ ClientNetworking::ClientNetworking() :
     m_io_service(),
     m_socket(m_io_service),
     m_incoming_messages(m_mutex),
-    m_connected(false),
-    m_cancel_retries(false)
+    m_rx_connected(false),
+    m_tx_connected(false)
 {}
 
-bool ClientNetworking::Connected() const {
+bool ClientNetworking::IsConnected() const {
     boost::mutex::scoped_lock lock(m_mutex);
-    return m_connected;
+    return m_rx_connected && m_tx_connected;
+}
+
+bool ClientNetworking::IsRxConnected() const {
+    boost::mutex::scoped_lock lock(m_mutex);
```

Comment:
@LGM-Doyle are you sure? I never thought a memory read operation not to be atomic on a low hardware level as this. If these low level memory access operations aren't atomic, how are OS level mutexes supposed to work? They need to store their states in memory too, and IIRC the lock, unlock, wait, signal etc. operations on semaphores&co need to be atomic.

It's been decades since these things have been hammered into my brain though, maybe my memory fails me here...